sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -937,6 +944,14 @@ private[execution] object AggregateProcessor {
       imperatives.toArray,
       trackPartitionSize)
   }
+
+  // Rewrites the window partition size attribute in all `SizeBasedWindowFunction`s using executor
+  // side `SizeBasedWindowFunction.n`.
+  def rewriteWindowPartitionSize(expression: Expression): Expression = {
+    expression.transformDown {
+      case f: SizeBasedWindowFunction => f.withPartitionSize(SizeBasedWindowFunction.n)
```

Comment:
I'd prefer not to do so since it doesn't bring much performance gain, yet depends on a not so obvious implicit assumption (partition size attribute ordinal is always 0) and makes the code harder to grasp.